SocketRocket/SRWebSocket.m

```diff
@@ -145,6 +145,122 @@ @implementation SRWebSocket {
     
     NSArray<NSString *> *_requestedProtocols;
     SRIOConsumerPool *_consumerPool;
+
+    //Proxy support
+    NSString *_httpProxyHost;
+    uint32_t _httpProxyPort;
+    BOOL _connectingToProxy;
+
+}
+
+// get proxy setting from device setting
+-(void) configureProxy
+{
+    SRFastLog(@"configureProxy");
+    NSDictionary *proxySettings = CFBridgingRelease(CFNetworkCopySystemProxySettings());
+    NSArray *proxies = CFBridgingRelease(CFNetworkCopyProxiesForURL((__bridge CFURLRef)_url, (__bridge CFDictionaryRef)proxySettings));
+    if (proxies.count == 0) {
+        SRFastLog(@"configureProxy no proxies");
+        [self openConnection];
+        return;                 // no proxy
+    }
+    NSDictionary *settings = [proxies objectAtIndex:0];
+    NSString *proxyType = settings[(NSString *)kCFProxyTypeKey];
+    if ([proxyType isEqualToString:(NSString *)kCFProxyTypeAutoConfigurationURL]) {
+        NSURL *pacURL = settings[(NSString *)kCFProxyAutoConfigurationURLKey];
+        if (pacURL) {
+            [self fetchPAC:pacURL];
+            return;
+        }
+    }
+    if ([proxyType isEqualToString:(NSString *)kCFProxyTypeHTTP] || [proxyType isEqualToString:(NSString *)kCFProxyTypeHTTPS]) {
+        _httpProxyHost = settings[(NSString *)kCFProxyHostNameKey];
+        _httpProxyPort = [settings[(NSString *)kCFProxyPortNumberKey] intValue];
+    }
+    if (_httpProxyHost) {
+        SRFastLog(@"SRWebSocket using proxy %@:%u", _httpProxyHost, _httpProxyPort);
+    }
+    [self openConnection];
+}
+
+- (void)fetchPAC:(NSURL *)PACurl
+{
+    SRFastLog(@"SRWebSocket fetchPAC:%@", PACurl);
+
+	if ([PACurl isFileURL]) {
+        NSError *nsError = nil;
+        NSString *script = [NSString stringWithContentsOfURL:PACurl
+                                                usedEncoding:NULL
+                                                       error:&nsError];
+
+        if (nsError) {
+            [self openConnection];
+            return;
+        }
+        [self runPACScript:script];
+		return;
```

Comment:
Ok,

I'll do it.

Dong

On 5/31/16 2:27 PM, Nikita Lutsenko wrote:
>
> In SocketRocket/SRWebSocket.m 
> <https://github.com/facebook/SocketRocket/pull/395#discussion_r65237082>:
>
> > +- (void)fetchPAC:(NSURL *)PACurl
> > +{
> > +    SRFastLog(@"SRWebSocket fetchPAC:%@", PACurl);
> > +
> > +	if ([PACurl isFileURL]) {
> > +        NSError *nsError = nil;
> > +        NSString *script = [NSString stringWithContentsOfURL:PACurl
> > +                                                usedEncoding:NULL
> > +                                                       error:&nsError];
> > +
> > +        if (nsError) {
> > +            [self openConnection];
> > +            return;
> > +        }
> > +        [self runPACScript:script];
> > +		return;
>
> Nit: Looks like this is missing indentation - if you could run 
> |auto-indent| in Xcode on all of the new code - that would help 
> tremendously.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/facebook/SocketRocket/pull/395/files/d1fc9a79282e2c2d1b28a928d131dc1b5947e986#r65237082>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AHRzfLTtowWQUdAdhjizU9S6dv5NZ3l0ks5qHH11gaJpZM4Ip_XE>.
>